### PR TITLE
fix: use default catalog toolbar for offset catalog table

### PR DESCRIPTION
**Pull request by drodil**
## Hey, I just made a Pull Request!

mostly to get debounce working the same way with the entity text filter

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by aramissennyeydd — APPROVED** [review]

**Comment by freben** [comment]
seems this needs a rebase now, sorry i've completely overlooked this one for so long, my bad

File: .changeset/curvy-poems-joke.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog': patch
+---
+
+Use default `CatalogTableToolbar` for offset paginated catalog table
```
Comment (vinzscam) [inline .changeset/curvy-poems-joke.md:5]:
```suggestion
Fixes an issue causing the `CatalogIndexPage` to not properly filter results when using offset pagination.
```

**Comment by vinzscam** [comment]
> seems this needs a rebase now, sorry i've completely overlooked this one for so long, my bad

Rebased and ready to go! I've added a suggestion to the changesets, as without this PR the table uses the native CatalogTable filtering which causes issues when searching groups via `spec.profile.displayName`.

**Review by freben — APPROVED** [review]

**Comment by tylerd-canva** [comment]
Nice! I was wondering about this. Should we add it to the non-paginated version in CatalogTable.tsx as well? I can do that in https://github.com/backstage/backstage/pull/27632

**Pull request merged**